### ISSUE TITLE
Grpc 1.10.1 C# may have some kind endless loop

### ISSUE THREAD

**Issue by luchunminglu** [grpc/grpc#15336]
### What version of gRPC and what language are you using?
 Grpc 1.10.1 C# from nuget
 
### What operating system (Linux, Windows, …) and version?
 windows server 2008 R2
 
### What runtime / compiler are you using (e.g. python version or version of gcc)
 .net 4.5
 
### What did you do?
If possible, provide a recipe for reproducing the error. Try being specific and include code snippets if helpful.

I Wirte a simple server. After about 25 days normal run, It first occurred in my client, which use 100% of one of my cpu core. Then after few days, it occurs in server which use 100% of all my cpu core.
pretty sure, the endless loop is not in my code.
The client first said Deadline Exceeded, after few hours, It block and never return.
The server cost 100% cpu, and never service any client.
After restart, it becomes normal.

```
var options = new List<ChannelOption>()
                {
                    new ChannelOption(ChannelOptions.MaxSendMessageLength,16*1024*1024),
                    new ChannelOption(ChannelOptions.MaxReceiveMessageLength,16*1024*1024),
                    new ChannelOption(ChannelOptions.MaxConcurrentStreams,63),
                    new ChannelOption(ChannelOptions.SoReuseport,1),
                };

                string ip = "0.0.0.0";
                
                _server = new Server(options)
                {
                    Services = { HealthCheckService.BindService(new HealthCheckServiceImpl()),MonitorService.BindService(new MonitorServiceImpl()),ElectService.BindService(new SelfElectServiceImpl()),ConfigService.BindService(new ConfigServiceImpl()),},
                    Ports = { new ServerPort(ip,serverPort,ServerCredentials.Insecure) }
                };

                _server.Start();
 

```
### What did you expect to see?
 Client and server works normal

**Comment by jtattermusch** [comment]
Thanks for reporting this. I'm not sure how to reproduce though - running a service for 25 days and hoping it's going to happen for me too doesn't seems like a good approach. Can you provide some additional info? (stacktrace, coredump, anything helps).

**Comment by apolcyn** [comment]
cc @sreecha - I think this sounds like the issue that is described and fixed in https://github.com/grpc/grpc/pull/14894

**Comment by jtattermusch** [comment]
An interesting observation:   25 days in milliseconds in ~2.1 billion milliseconds  which is pretty close to  max_int for 32bit integers - 2,147,483,647 - and we are using milliseconds since process start internally to deal with deadlines in C core.

@luchunminglu to confirm this theory, are you running a 32bit version of the binary?  (if overflow is the issue, this would only happen with 32bit and 64bit would run normally).

**Comment by jtattermusch** [comment]
@apolcyn  you were faster than me, I think this is really an instance of https://github.com/grpc/grpc/pull/14894

**Comment by luchunminglu** [comment]
I run this as 32bit application on a 64bit OS

**Comment by luchunminglu** [comment]
Pure Client seems work fine, It occured at server application and server&client application（it have a grpc server but also as a client to other grpc server）

**Comment by jtattermusch** [comment]
I think this was fixed by https://github.com/grpc/grpc/pull/14894. Fix will be in the next release (v1.13.0).
